static/app/views/dashboardsV2/widgetBuilder/buildSteps/sortByStep/index.tsx

```diff
@@ -62,7 +73,11 @@ export function SortByStep({
               <ResultsLimitSelector
                 name="resultsLimit"
                 menuPlacement="auto"
-                options={[...Array(RESULTS_LIMIT).keys()].map(resultLimit => {
+                options={[
+                  ...Array(
+                    getResultsLimit(queries.length, queries[0].aggregates.length)
```

Comment:
since we are using the same function inside of the useEffect too, what do you think we move it to the very top and add the result as a dependency of effect? and we can reuse the results here too